src/argilla/client/feedback/schemas/records.py

```diff
@@ -232,41 +263,129 @@ class RemoteFeedbackRecord(FeedbackRecord):
+
+            if suggestion.question_name in new_suggestions:
+                warnings.warn(
+                    f"A suggestion for question `{suggestion.question_name}` has been"
+                    " provided twice in the same update, so the last one will be the one"
+                    " to be kept.",
+                    UserWarning,
+                    stacklevel=1,
+                )
+                new_suggestions.pop(suggestion.question_name, None)
+                new_suggestions[suggestion.question_name] = suggestion
+            elif suggestion.question_name in existing_suggestions:
+                comparable_fields = {"question_name", "type", "score", "value", "agent"}
+                comparable_suggestion = suggestion.dict(include={"question_name", "type", "score", "value", "agent"})
+                if any(
+                    [
+                        comparable_suggestion == suggestion.dict(include=comparable_fields)
+                        for suggestion in existing_suggestions.values()
+                    ]
+                ):
+                    warnings.warn(
+                        f"A suggestion for question `{suggestion.question_name}` has already"
+                        " been provided and the provided suggestion is the same, so it will"
+                        " be ignored.",
+                        UserWarning,
+                        stacklevel=1,
+                    )
+                else:
+                    warnings.warn(
+                        f"A suggestion for question `{suggestion.question_name}` has already"
+                        " been provided but the provided suggestion is different, so it will"
+                        " overwrite the existing one.",
+                        UserWarning,
+                        stacklevel=1,
+                    )
+                    existing_suggestions.pop(suggestion.question_name, None)
+                    new_suggestions[suggestion.question_name] = suggestion
+            else:
+                new_suggestions[suggestion.question_name] = suggestion
+
+        for suggestion in new_suggestions.values():
+            if isinstance(suggestion, SuggestionSchema):
+                exclude = {"question_name"}
+            elif isinstance(suggestion, RemoteSuggestionSchema):
+                exclude = {"client", "id", "record_id", "question_name"}
+            pushed_suggestion = datasets_api_v1.set_suggestion(
+                client=self._client, record_id=self.id, **suggestion.dict(exclude_none=True, exclude=exclude)
+            )
+            existing_suggestions[suggestion.question_name] = RemoteSuggestionSchema(
+                client=self._client,
+                record_id=self.id,
+                question_name=suggestion.question_name,
+                **pushed_suggestion.parsed.dict(exclude_none=True),
             )
 
+        self.__dict__["suggestions"] = tuple(existing_suggestions.values())
+
```

Comment:
I'd say this is not required at the moment, since the `update` method just updates the `suggestions` at this point, for sure something to do later, but not priority now